climpyrical/tests/test_gridding.py

```diff
@@ -34,72 +36,72 @@ def test_check_ndims(data, n, passed, error):
-xi = np.linspace(-33.8800048828125, 33.8800048828125, 155)
-yi = np.linspace(-28.59999656677246, 28.15999984741211, 130)
-xext, yext = np.tile(xi, yi.size), np.repeat(yi, xi.shape)
-xext_bad, yext_bad = np.repeat(xi, yi.size), np.tile(yi, xi.shape)
+# load example ensemble dataset for testing
+ds = read_data(
+        resource_filename("climpyrical", "tests/data/snw_test_ensemble.nc"),
+        "Rain-RL50"
+    )
+
+# read grids with expected dimension and ranges
+xi, yi = ds.rlon.values, ds.rlat.values
+
+# extend coordinates in expected way
+xext_ex, yext_ex = np.tile(xi, yi.size), np.repeat(yi, xi.size)
+xext_bad, yext_bad = np.repeat(xi, yi.size), np.tile(yi, xi.size)
 
 
 @pytest.mark.parametrize(
-    "x,y,passed,error",
+    "x,y,ds,passed,error",
     [
-        ("x", np.linspace(-24, 24, 155), False, TypeError),
-        (np.linspace(-24, 24, 155), "y", False, TypeError),
-        (
-            np.linspace(0, 10, 30),
-            np.linspace(-28.59999656677246, 28.15999984741211, 130),
-            False,
-            ValueError,
-        ),
-        (
-            np.linspace(-33.8800048828125, 33.8800048828125, 155),
-            np.linspace(0, 10, 130),
-            False,
-            ValueError,
-        ),
-        (xi, yi, True, None),
-        (xext, yext, True, None),
+        ("x", yi, ds, False, TypeError),
+        (xi, "y", ds, False, TypeError),
+        (np.linspace(0, 10, 30), yi, ds, False, ValueError),
+        (xi, np.linspace(0, 10, 130), ds, False, ValueError),
+        (xi, yi, ds, True, None),
+        (xext_ex, yext_ex, ds, True, None),
     ],
 )
-def test_check_input_coords(x, y, passed, error):
+def test_check_input_coords(x, y, ds, passed, error):
     if passed:
-        check_input_coords(x, y)
+        check_input_coords(x, y, ds)
     else:
         with pytest.raises(error):
-            check_input_coords(x, y)
+            check_input_coords(x, y, ds)
 
 
 @pytest.mark.parametrize(
-    "x,y,xext,yext,passed",
+    "x,y,xext_ex,yext_ex,ds,passed,error",
     [
-        (xi, yi, xext, np.delete(yext, xi.size), False),
-        (xi, np.delete(yi, 2), xext, yext, False),
-        (xi, yi, xext, yext, True),
-        (xi, yi, xext_bad, yext, False),
-        (xi, yi, xext, yext_bad, False),
-        (yi, xi, xext, yext, False),
-        (yi, xi, yext, xext, False),
+        (xi, yi, xext_ex, np.delete(yext_ex, xi.size), ds, False, ValueError),
+        (xi, np.delete(yi, 2), xext_ex, yext_ex, ds, False, ValueError),
+        (xi, yi, xext_ex, yext_ex, ds, True, None),
+        (xi, yi, xext_bad, yext_ex, ds, False, ValueError),
+        (xi, yi, xext_ex, yext_bad, ds, False, ValueError),
+        (yi, xi, xext_ex, yext_ex, ds, False, ValueError),
+        (yi, xi, yext_ex, xext_ex, ds, False, ValueError),
     ],
 )
-def test_check_coords_are_flattened(x, y, xext, yext, passed):
+def test_check_coords_are_flattened(x, y, xext_ex, yext_ex, ds, passed, error):
```

Comment:
Minor point: Param `passed` is redundant: you can naturally use the value `None` for param `error` to denote the passed condition.